ote/src/cljs/ote/app/controller/service_search.cljs

```diff
@@ -68,4 +72,25 @@
     (let [{:keys [empty-filters? results]} response]
       (update app :service-search assoc
               :results results
-              :empty-filters? empty-filters?))))
+              :empty-filters? empty-filters?)))
+
+  ShowServiceGeoJSON
+  (process-event [{:keys [url]} app]
+    (comm/get! "viewer" {:params {:url url}
+                         :on-success (tuck/send-async! ->GeoJSONFetched)
+                         :response-format :json})
+    (update app :service-search assoc
+            :resource nil
+            :geojson nil
+            :loading-geojson? true))
+
+  CloseServiceGeoJSON
+  (process-event [_ app]
+    (update app :service-search dissoc :resource :geojson :loading-geojson?))
+
+  GeoJSONFetched
+  (process-event [{response :response} app]
+    (update app :service-search assoc
+            :resource response
```

Comment:
:resource and :geojson keys both have the whole response as value.
Should this be optimised somehow?

Reply:
The geojson (as a JS object) is required for Leaflet to draw it